site/zenodo_rdm/legacy/serializers/schemas.py

```diff
@@ -163,6 +219,64 @@ def dump_resource_type(self, result, original, **kwargs):
                 result[f"{upload_type}_type"] = resource_type_id.split("-")[-1]
         return result
 
+    @post_dump(pass_original=True)
+    def dump_subjects(self, result, original, **kwargs):
+        """Dumps subjects."""
+        subjects = original.get("subjects", [])
+        serialized_subjects = []
+        serialized_keywords = []
+        if subjects:
+            for _sbj in subjects:
+                _id = _sbj.get("id")
+                _subject = _sbj.get("subject")
+                # If subject has an id, it's a controlled vocabulary
+                if _id:
+                    # TODO we still did not define a strategy to map legacy subjects to rdm.
+                    pass
+                # Otherwise it's a free text string (keyword)
+                elif _subject:
+                    serialized_keywords.append(_subject)
+
+        if serialized_keywords:
+            result["keywords"] = serialized_keywords
+
+        if serialized_subjects:
+            result["subjects"] = serialized_subjects
+
+        return result
+
+    def dump_reference(self, obj):
+        """Dumps reference."""
+        references_list = obj.get("references", [])
+        if not references_list:
+            return missing
+
+        return [_ref["reference"] for _ref in references_list]
+
+    def dump_access_right(self, obj):
+        """Dumps access right."""
+        access = obj["access"]
+        files_access = access["files"]
+        is_open = files_access == "public"
+        is_embargoed = access.get("embargo", {}).get("active")
+        is_restricted = not is_embargoed and files_access == "restricted"
+
+        legacy_access = None
+
+        if is_open:
+            legacy_access = "open"
+        elif is_restricted:
+            legacy_access = "restricted"
+        elif is_embargoed:
+            legacy_access = "embargoed"
+        # TODO how to map to closed access?
```

Comment:
Currently we can deserialize access `closed` (legacy) to `files->restricted` (RDM).
However we are not able to serialize it back.

Reply:
minor: actually that's `closed` access in legacy, since `restricted` is a special case of `closed` for which we need the "Access requests" feature to be implemented).

Reply:
I am not sure what you meant by that, sorry 😅  
For now I dropped a TODO here to signal that we need to come back to this after access requests are implemented